File: patterns/java/src/refactoring_guru/patterns/factory_method/example/factory/HtmlDialog.java
package refactoring_guru.patterns.factory_method.example.factory;

import refactoring_guru.patterns.factory_method.example.buttons.Button;
import refactoring_guru.patterns.factory_method.example.buttons.HtmlButton;

/**
 * EN: HTML Dialog will produce HTML buttons.
 */
public class HtmlDialog extends Dialog {

    @Override
    public Button createButton() {
        return new HtmlButton();
    }
}

File: patterns/cs/src/RefactoringGuru/FactoryMethod/Example/Factory/HtmlDialog.cs
using RefactoringGuru.FactoryMethod.Example.Buttons;

namespace RefactoringGuru.FactoryMethod.Example.Factory
{
    /**
     * EN: HTML Dialog will produce HTML buttons.
     * 
     * RU: HTML-диалог.
     */
    class HtmlDialog : Dialog
    {
        public override IButton CreateButton()
        {
            return new HtmlButton();
        }
    }
}
